extensions/walletProviders/kadena/src/extension.ts

```diff
@@ -0,0 +1,79 @@
+import { ITatumSdkContainer, Network, TatumConfig, TatumSdkWalletProvider } from '@tatumio/tatum'
+import { KadenaNetworkId, KadenaTxPayload, KadenaWallet } from './types'
+import { genKeyPair } from "@kadena/cryptography-utils/lib/genKeyPair";
+import { restoreKeyPairFromSecretKey } from "@kadena/cryptography-utils";
+import { ChainId, createClient, createSignWithKeypair, ICommand } from "@kadena/client";
+import { ICommandPayload } from "@kadena/types";
+import { KadenaLoadBalancerRpc } from "@tatumio/tatum/dist/src/service/rpc/other/KadenaLoadBalancerRpc";
+
+export class KadenaWalletProvider extends TatumSdkWalletProvider<KadenaWallet, KadenaTxPayload> {
+  private readonly kadenaRpc: KadenaLoadBalancerRpc
+  private readonly sdkConfig: TatumConfig
+
+  constructor(tatumSdkContainer: ITatumSdkContainer) {
+    super(tatumSdkContainer)
+    this.sdkConfig = this.tatumSdkContainer.getConfig()
+    this.kadenaRpc = this.tatumSdkContainer.get(KadenaLoadBalancerRpc)
+  }
+
+  /**
+   * Gets a Kadena wallet, which includes an address, secret key, and a public key.
+   * @returns {KadenaWallet} An object containing address, secret key, and public key.
+   */
+  public async getWallet() {
+    const keyPair = genKeyPair()
+    const address = 'k:' + keyPair.publicKey
+
+    return { address, publicKey: keyPair.publicKey, secretKey: keyPair.secretKey} as KadenaWallet
+  }
+
+  /**
+   * Signs and broadcasts a Kadena transaction payload.
+   * @param {KadenaTxPayload} txPayload - The Kadena transaction payload, which includes secret key and transaction details.
+   * @returns {Promise<string>} A promise that resolves to the transaction hash.
+   */
+  public async signAndBroadcast(txPayload: KadenaTxPayload): Promise<string> {
+    const keyPair = restoreKeyPairFromSecretKey(txPayload.secretKey)
+    const signWithKeypair = createSignWithKeypair(keyPair)
+    const payload: ICommandPayload = JSON.parse(txPayload.command.cmd)
+
+    const chainId = payload.meta.chainId
+
+    if(!chainId)
+    {
+      throw new Error('Chain ID is required to sign and broadcast a Kadena transaction.')
+    }
+
+    const { client} = this.getTatumKadenaClient(chainId);
+
+    const signedCommand = await signWithKeypair(txPayload.command);
+    const result = await client.submit(signedCommand as ICommand)
+    return result.requestKey
+  }
+
+  private getTatumKadenaClient(chainId: ChainId) {
+    const networkId = this.getKadenaNetworkId();
+    const nodeUrl = this.kadenaRpc.getRpcNodeUrl()
+    const host = `${nodeUrl}/chainweb/0.0/${networkId}/chain/${chainId}/pact`
+    console.log(`host: ${host}`)
```

Comment:
Did you mean to keep the `console.log` in? Just making sure since the SDK has its own logger, but I am not sure how usable it is with extensions tbh.

Reply:
Removed